### ISSUE TITLE
ReferenceError: navigator is not defined

### ISSUE THREAD

**Issue by chmac** [Xabadu/react-native-jest-mocks#11]
I've added this package to my dev dependencies, I've imported it and run `mock.initAll()` in my test setup file, but I still see the following error:

    ReferenceError: navigator is not defined

I'm accessing geolocation like `navigator.geolocation.requestAuthorization()`.

I had a look at the [mock-geolocatin implementation](https://github.com/2gis/mock-geolocation/blob/master/src/geolocate.js) and it looks like it specifically targets the `navigator` global. Maybe the solution is to do something similar here?

**Comment by Xabadu** [comment]
Hey @chmac, thanks for opening this issue. The whole navigator mock definitely needs some extra work to make it more complete.

Quick question: What version of RN are you using? I remember that when I added the `Geolocation` mock I was using `navigator.geolocation.getCurrentPosition` in my app, and the navigator global was injected by default, so I'm wondering if maybe something changed between versions.

Thanks for adding the link as well. I'm gonna dig into it to fix this. If you come up with a solution in the meantime, feel free to open a PR 👍

**Comment by chmac** [comment]
I spent a little time on it, but I couldn't get it working. I added `navigator` to the `globals` property of the jest settings in `package.json` so then the `navigator` is created, but as I read the code in this package, it's mocking an import called `Geolocation`. I'm using the latest stable react native (`0.50.3`). I wondered if the API used to be `import foo from Geolocation` and so the mock would work as `jest.mock('Geolocation', ...)` but to be honest, this is my first experiment with jest, so I really don't know much about it.

I tried naively changing it to `jest.mock('navigator.geolocation`, ...)` but that didn't seem to work. I'm not sure if `jest.mock()` is setup for that, I have a feeling that `navigator.geolocation` needs to be replaced with a mock, but I could be wrong. I also look a bit at [mock-geolocation](https://github.com/2gis/mock-geolocation). It only works in the browser, but it provides a nice API to mock different scenarios. It's hard coded around `window.navigator`, but perhaps the author is open to supporting React Native, and maybe there's a way to integrate it into your package somehow.

EDIT: [This thread](https://github.com/2gis/mock-geolocation/issues/2) could be useful.
